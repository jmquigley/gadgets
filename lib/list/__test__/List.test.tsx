'use strict';

import * as assert from 'assert';
import {mount, shallow} from 'enzyme';
import * as React from 'react';
import {getDefaultListProps, List, ListItem} from '../index';

test('Test retrieval of List props object', () => {
	const props = getDefaultListProps();

	assert(props);
	expect(props).toMatchSnapshot();
});

test('Test the creation of a List control container', () => {
	const ctl = shallow(
		<List alternating className="test-class" testing>
			<li>some list item</li>
		</List>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test disabling of a List control', () => {
	const ctl = shallow(
		<List disabled={true} testing>
			<li>some list item</li>
		</List>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test making List control invisible', () => {
	const ctl = shallow(
		<List visible={false} testing>
			<li>some list item</li>
		</List>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test a list with ListItem and selection', () => {
	const ctl = mount(
		<List testing>
			<ListItem title="Item #1" />
			<ListItem title="Item #2" />
		</List>
	);

	assert(ctl);

	assert(!ctl.prop('disabled'));
	assert(ctl.prop('visible'));
	assert.equal(ctl.find('.ui-list').length, 1);

	const li1 = ctl.find(ListItem).first();
	const li2 = ctl.find(ListItem).last();
	assert.equal(li1.text(), 'Item #1');
	assert.equal(li2.text(), 'Item #2');
});
