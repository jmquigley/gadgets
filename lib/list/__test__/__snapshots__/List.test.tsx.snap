// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test disabling of a List control 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "color": "#000000",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "tooltipBackgroundColor": "gray",
      "tooltipForegroundColor": "white",
      "transitionDelay": "0.5s",
    }
  }
>
  <styled.ul
    alternating={false}
    className="ui-list"
    style={Object {}}
  >
    <li
      href={
        Object {
          "selectHandler": [Function],
          "sizing": "normal",
        }
      }
      id="0"
      key=".0"
      selected={false}
    >
      some list item
    </li>
  </styled.ul>
</ThemeProvider>
`;

exports[`Test making List control invisible 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "color": "#000000",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "tooltipBackgroundColor": "gray",
      "tooltipForegroundColor": "white",
      "transitionDelay": "0.5s",
    }
  }
>
  <styled.ul
    alternating={false}
    className="ui-list"
    style={Object {}}
  >
    <li
      href={
        Object {
          "selectHandler": [Function],
          "sizing": "normal",
        }
      }
      id="0"
      key=".0"
      selected={false}
    >
      some list item
    </li>
  </styled.ul>
</ThemeProvider>
`;

exports[`Test retrieval of List props object 1`] = `
Object {
  "alternating": false,
  "children": null,
  "className": "",
  "contentEditable": false,
  "direction": "right",
  "disabled": false,
  "focus": false,
  "height": "",
  "id": "",
  "location": "none",
  "noedit": false,
  "nohover": false,
  "noripple": false,
  "onAdd": [Function],
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": null,
  "tooltip": "",
  "unselect": false,
  "visible": true,
  "width": "",
  "xcss": null,
}
`;

exports[`Test the creation of a List control container 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "color": "#000000",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "tooltipBackgroundColor": "gray",
      "tooltipForegroundColor": "white",
      "transitionDelay": "0.5s",
    }
  }
>
  <styled.ul
    alternating={true}
    className="ui-list test-class"
    style={Object {}}
  >
    <li
      href={
        Object {
          "selectHandler": [Function],
          "sizing": "normal",
        }
      }
      id="0"
      key=".0"
      selected={false}
    >
      some list item
    </li>
  </styled.ul>
</ThemeProvider>
`;
